plugins/cache-ehcache/src/main/java/org/opensearch/cache/EhcacheDiskCacheSettings.java

```diff
@@ -110,6 +111,22 @@ public class EhcacheDiskCacheSettings {
         (key) -> Setting.boolSetting(key, false, NodeScope)
     );
 
+    /**
+     * Defines whether to use an in-memory keystore to check for probable presence of keys before having to go to disk.
+     */
+    public static final Setting.AffixSetting<Boolean> USE_RBM_KEYSTORE_SETTING = Setting.suffixKeySetting(
+        EhcacheDiskCache.EhcacheDiskCacheFactory.EHCACHE_DISK_CACHE_NAME + "use_keystore",
+        (key) -> Setting.boolSetting(key, true, NodeScope)
+    );
```

Comment:
Should be `.use_keystore`, note the dot prefix. Same for below.

Also, Instead of defining like above should we have something like below?

```
public static final Setting.AffixSetting<String> USE_RBM_KEYSTORE_SETTING = Setting.suffixKeySetting(
        EhcacheDiskCache.EhcacheDiskCacheFactory.EHCACHE_DISK_CACHE_NAME + "keystore",
        (key) -> Setting.simpleStringSetting(key, "rbm", NodeScope)
    );
```

Here we mention "which" keystore to use instead of having a setting explicitly on RBM.
"rbm" would be one of the implementation
And we can set this to null or "" if we don't want to use it. In this case we will create a mock keyStore which does nothing.


This way we can eventually have different kind of keystores if needed in the future and load them from respective plugins(if needed). As of now we can keep this inside the same plugin.

@sohami What do you think?

Reply:
Good catch on the dot prefix!